plugins/pi4j-plugin-pigpio/src/main/java/com/pi4j/plugin/pigpio/provider/i2c/PiGpioI2C.java

```diff
@@ -150,6 +150,19 @@ public int writeRegister(int register, byte[] data, int offset, int length) {
         return length;
     }
 
+    @Override
+    /**
+     * {@inheritDoc}
+     * <p> Note: Function not supported with PIGPIO provider.
+     * This method 'is' supported in the LinuxFS provider
+     * </p>
+     */
+    public int writeRegister(byte[] register, byte[] data, int offset, int length) {
+        // TODO should we throw an exception,  saying use LinuxFS ?  What exception
+        // TODO any what to force a compile error in user code ??
+        return -1;
```

Comment:
Yes, i think we should throw new IllegalStateException("Not supported, please use LinuxFS plugin")

Reply:
Thanks sirs.  I will add that exception as the answer to the added interface showing in pigpio but not being supported 

Will cleanup  up neglected format and the specific  code comments. 

@eitch @FDelporte